qiskit/extensions/hamiltonian_gate.py

```diff
@@ -113,6 +114,14 @@ def qasm(self):
         """Raise an error, as QASM is not defined for the HamiltonianGate."""
         raise ExtensionError("HamiltonianGate as no QASM definition.")
 
+    def validate_parameter(self, parameter):
+        """Hamiltonian parameter has to be an ndarray, operator or float."""
+        if isinstance(parameter, (float, int, numpy.ndarray)):
```

Comment:
Will this cover `Operator`s? From https://github.com/Qiskit/qiskit-aqua/blob/17c21272bce616918f021e90155fcc61b4b8818e/qiskit/aqua/operators/primitive_ops/matrix_op.py#L165 , it looks like there's a broader list that had been supported here: https://github.com/Qiskit/qiskit-aqua/blob/17c21272bce616918f021e90155fcc61b4b8818e/qiskit/aqua/operators/primitive_ops/primitive_op.py#L51

Reply:
As far as I understand, `__init__` convert these parameters to matrices before calling super...  Any example?

Reply:
Indeed it will!